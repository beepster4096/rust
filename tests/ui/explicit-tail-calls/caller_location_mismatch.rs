#![feature(explicit_tail_calls)]
use std::panic::Location;

fn get_some_location1() -> &'static Location<'static> {
    #[track_caller]
    fn inner() -> &'static Location<'static> {
        become Location::caller()
    }

    become inner()
    //~^ error: a function mot marked with `#[track_caller]` cannot tail-call one that is
}

#[track_caller]
fn get_some_location2() -> &'static Location<'static> {
    fn inner() -> &'static Location<'static> {
        become Location::caller()
        //~^ error: a function mot marked with `#[track_caller]` cannot tail-call one that is
    }
    
    become inner()
    //~^ error: a function marked with `#[track_caller]` cannot tail-call one that is not
}

fn main() {
    get_some_location1();
    get_some_location2();
}
