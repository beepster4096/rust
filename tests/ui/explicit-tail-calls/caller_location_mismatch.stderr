error: a function mot marked with `#[track_caller]` cannot tail-call one that is
  --> $DIR/caller_location_mismatch.rs:10:5
   |
LL |     become inner()
   |     ^^^^^^^^^^^^^^

error: a function marked with `#[track_caller]` cannot tail-call one that is not
  --> $DIR/caller_location_mismatch.rs:21:5
   |
LL |     become inner()
   |     ^^^^^^^^^^^^^^

error: a function mot marked with `#[track_caller]` cannot tail-call one that is
  --> $DIR/caller_location_mismatch.rs:17:9
   |
LL |         become Location::caller()
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

